### DISCUSSION TITLE
Visual Editing?

### DISCUSSION THREAD

**Discussion by mbrain-io** [Thinkmill/keystatic#311] [Ideas]
Hi there, awesome work! Is there a plan to introduce some kind of visual editing? For example ability to click text on the page and directly edit the Keystatic data? Or even ability to define section blocks that can be added to the page and visually add/edit those blocks?

**Comment by simonswiss** [comment]
Hey!

The first part (clicking on text directly in the UI and start editing), no plans of doing this at this point.

However, I _think_ what you mean by the second point is aligned with what the team is working on. You can definitely [define sections (called `componentBlocks`) for the `document` field](https://keystatic.com/docs/fields/document), and have a visual representation of those in the Admin UI.

This screenshot is a bit outdated, but should give you an idea of what I mean by that:

<img width="1657" alt="CleanShot 2023-06-14 at 15 07 05@2x" src="https://github.com/Thinkmill/keystatic/assets/485747/ea824139-8b5a-4892-b99f-939c811df9fc">

Here's a more recent take on the UI:

<img width="987" alt="CleanShot 2023-06-14 at 15 14 46@2x" src="https://github.com/Thinkmill/keystatic/assets/485747/e2b82969-689f-4cfe-8670-7b0091da87d4">

Hope it helps!

**Comment by mbrain-io** [comment]
@simonswiss So we can visually see the components we are adding to the document, thats awesome! I only hope that the button used to add those components to the document will allow adding a lot of them. I.e. that it will have some visual component preview, component grouping and component search mechanism.
